test/test_helpers.h

```diff
@@ -11,5 +11,6 @@
 #include <float.h>
 
 #define assert_double_equal(a, b) assert_true(fabs(a - b) <= DBL_EPSILON)
+#define assert_float_equal(a, b) assert_true(fabs(a - b) <= FLT_EPSILON)
```

Comment:
not to add another lightbulb screwer inner, but the bound should properly be `FLT_EPSILON * fmax(fabs(a),fabs(b))` as with `assert_double equal`

Reply:
actually for maximum correctness, should us `fabsf` / `fmaxf` for the `float` vairant

Reply:
I get `fabsf` correction, but why do `FLT_EPSILON * fmax(fabs(a),fabs(b))`?

Reply:
Because floating point error is relative.
1.00000000000000022 - 1.0 = 2.2e-16 ~ 0
1.00000000000000022e7 - 1.0e7 = 2.2e-9 ~ 0